boards/imix/src/icmp_lowpan_test.rs

```diff
@@ -147,7 +147,7 @@ impl<'a, A: time::Alarm> capsules::net::icmpv6::icmpv6_send::ICMP6SendClient
     }
 }
 
-impl<'a, A: time::Alarm + 'a> LowpanICMPTest<'a, A> {
+impl<'a, A: time::Alarm> LowpanICMPTest<'a, A> {
```

Comment:
Why can't we remove the first `'a` here? (and numerous other places such as ipv6_lowpan_test.rs, sixlowpan_dummy.rs, udp_lowpan_test.rs, and traitobj_list.rs)

Reply:
I think we can. Commits welcome :) (It's a test file so my motivation was low).

Reply:
First try doesn't "just work":

```
   Compiling imix v0.1.0 (file:///Volumes/code/helena-project/tock/boards/imix)
error[E0106]: missing lifetime specifier
   --> src/icmp_lowpan_test.rs:150:22
    |
150 | impl<A: time::Alarm> LowpanICMPTest<A> {
    |                      ^^^^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/icmp_lowpan_test.rs:157:10
    |
157 |     ) -> LowpanICMPTest<A> {
    |          ^^^^^^^^^^^^^^^^^ expected lifetime parameter
    |
    = help: this function's return type contains a borrowed value, but the signature does not say which one of `icmp_sender`'s 2 lifetimes it is borrowed from

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0106`.
```

Messing around a bit I couldn't puzzle out any other thing that would make the compiler happy, but I didn't try too hard